source/crud/bulk-write.md

```diff
@@ -0,0 +1,724 @@
+     */
+    collation: Optional<Document>;
+
+    /**
+     * The index to use. Specify either the index name as a string or the index key pattern. If
+     * specified, then the query system will only consider plans using the hinted index.
+     *
+     * This option is only sent if the caller explicitly provides a value.
+     */
+    hint: Optional<(String | Document)>;
+}
+```
+
+Each write model provided to `MongoClient.bulkWrite` in the `models` parameter MUST have a
+corresponding namespace that defines the collection on which the operation should be performed.
+Drivers SHOULD design this pairing in whichever way is most ergonomic for its language. For
+example, drivers may:
+
+- Include a required `namespace` field on each `WriteModel` variant.
+- Accept a list of `(Namespace, WriteModel)` tuples for `models`.
+- Define the following pair class:
+
+```typescript
+class NamespaceWriteModelPair {
+    /**
+     * The namespace on which to perform the write.
+     */
+    namespace: Namespace;
+
+    /**
+     * The write to perform.
+     */
+    model: WriteModel;
+}
+```
+
+Drivers MUST throw an exception if the list provided for `models` is empty.
+
+### Options
+
+```typescript
+class BulkWriteOptions {
+    /**
+     * Whether the operations in this bulk write should be executed in the order in which they were
+     * specified. If false, writes will continue to be executed if an individual write fails. If
+     * true, writes will stop executing if an individual write fails.
+     *
+     * Defaults to false.
+     */
+    ordered: Optional<Boolean>;
+
+    /**
+     * If true, allows the writes to opt out of document-level validation.
+     *
+     * Defaults to false.
+     *
+     * This option is only sent if the caller explicitly provides a value.
+     */
+    bypassDocumentValidation: Optional<Boolean>;
+
+    /**
+     * A map of parameter names and values to apply to all operations within the bulk write. Value
+     * must be constant or closed expressions that do not reference document fields. Parameters can
+     * then be accessed as variables in an aggregate expression context (e.g. "$$var").
+     *
+     * This option is only sent if the caller explicitly provides a value.
+     */
+    let: Optional<Document>;
+
+    /**
+     * The write concern to use for this bulk write.
+     *
+     * NOT REQUIRED TO IMPLEMENT. Drivers MUST expose this option if retrieving a handle to a
+     * client with a different write concern configured than that of the user's standard URI
+     * options is nontrivial. Drivers MAY omit this option if they provide a way to retrieve a
+     * lightweight handle to a client with a custom write concern configured, e.g. a
+     * MongoClient.withWriteConcern() method.
+     */
+    writeConcern: Optional<WriteConcern>;
+
+    /**
+     * Whether detailed results for each successful operation should be included in the returned
+     * BulkWriteResult.
```

Comment:
It may be helpful to note that this corresponds inversely to the value of `errorsOnly` in the `bulkWrite` command. Not necessary for other options since those have a direct counterpart.